lib/seahorn/Smt/MarshalYices.cc

```diff
@@ -262,7 +262,13 @@ term_t marshal_yices::encode_term(Expr e, ycache_t &cache) {
     if (isOpX<REM>(e)) {
       encode_term_fail(e, "Integer remainder not supported in yices");
     } else if (isOpX<CONST_ARRAY>(e)) {
-      encode_term_fail(e, "const-array term not supported in yices");
+      // (const-array sort v) --> lambda i:sort :: v
+      // idea from:
+      // https://github.com/SRI-CSL/yices2/issues/271#issuecomment-664726905
+      type_t indexType = encode_type(e->left());
+      term_t i = yices_new_variable(indexType);
+      term_t v = encode_term(e->right(), cache);
+      return yices_lambda(1, &i, v);
```

Comment:
because of the early return, you are missing caching. This supposed to assign `res` variable like the surrounding code, and then at the end of the function `res` is cached.

Reply:
Maybe do early caching here? I don't like the idea of going through rest of the `else if` blocks and do type checking.

Reply:
structure the code so that caching happens but of course other branches should not be evaluated